packages/mermaid/src/docs/index.md

```diff
@@ -32,6 +32,17 @@ features:
 
 <script setup>
 import { VPTeamMembers } from 'vitepress/theme'
+import { getRedirect } from './.vitepress/redirect';
+
+try{
+  const newPath = getRedirect(window.location);
+  if(newPath) {
+    console.log(`Redirecting to ${newPath} from ${window.location}`);
+    window.location.replace(`/mermaid/${getRedirect(window.location)}.html`);
+  }
+}catch(e){
+}
+
```

Comment:
It might be impossible, but a nicer way of doing this would be to modify VitePress's router, but I couldn't figure out if there was an easy way of doing this:

```js
import DefaultTheme from 'vitepress/theme'

export default {
  ...DefaultTheme,
  enhanceApp(ctx) {
    DefaultTheme.enhanceApp(ctx)
    
    // can we do something with this?
    // I can't find anything in the API, see https://github.com/vuejs/vitepress/blob/main/src/client/app/router.ts
    const {router} = ctx;
  }
}
```

It could be worth making a feature request issue on VitePress for this.

Reply:
Done :)